### PR TITLE
[DLS-6533] Adding view to return SentController and flag for submission

### PR DESCRIPTION
**Pull request by JakeReid2020**

### CODE REVIEW

File: app/controllers/ReturnSentController.scala [hunk 1]
```diff
@@ -16,30 +16,211 @@
 
 package controllers
 
+import config.FrontendAppConfig
```
Comment (RaminEsfandiari) [inline app/controllers/ReturnSentController.scala:19]: Could you optimise the imports please?

image

File: app/controllers/ReturnSentController.scala [hunk 2]
```diff
@@ -16,30 +16,211 @@
...
+  //Warehouse
+  val tradingName: String = "Soft Juice Ltd"
+  val line1: String = "3 Prospect St"
+  val line2: String = "Reading"
+  val line3: String = "Berkshire"
+  val line4: String = "United Kingdom"
+  val postcode: String = "CT44 0DF"
+  val warehouseList: List[Warehouse] = List(Warehouse(tradingName, Address(line1, line2, line3, line4, postcode)))
+  val logger: Logger = Logger(this.getClass())
+
```
Comment (RaminEsfandiari) [inline app/controllers/ReturnSentController.scala:61]: I assume we are reverting back to what we have pre-cya to allow the refactor to go ahead so we are leaving these hardcoded ones in.

File: test/controllers/ReturnSentControllerSpec.scala [hunk 3]
```diff
@@ -17,30 +17,550 @@
...
+class ReturnSentControllerSpec extends SpecBase with SummaryListFluency with BeforeAndAfter {
 
   "ReturnSent Controller" - {
 
-    "must show returns sent page" in {
+    val zero = BigDecimal(0.00)
+    val amounts = Amounts(zero, zero, zero)
+    val mockSessionCache = mock[SDILSessionCache]
+    when(mockSessionCache.fetchEntry[Amounts](any(), any())(any())) thenReturn Future.successful(Some(amounts))
+    val returnPeriodsContainingBaseReturnPeriod = List(ReturnPeriod(2020, 0), ReturnPeriod(2023, 1), returnPeriod)
```
Comment (RaminEsfandiari) [inline test/controllers/ReturnSentControllerSpec.scala]: This is not used, can be removed. You can also remove ReturnPeriod from the imports.

File: test/controllers/ReturnSentControllerSpec.scala [hunk 4]
```diff
@@ -17,30 +17,550 @@
...
+
+      running(application) {
+        val request = FakeRequest(GET, routes.ReturnsController.onPageLoad(false).url)
+        val result = route(application, request).value
+
+        status(result) mustEqual OK
+      }
+    }
+
+  }
```
Comment (RaminEsfandiari) [inline test/controllers/ReturnSentControllerSpec.scala:564]: Tests are barely covering controller code. I understand the direction has been set to get a returns sent page in for the refactoring to continue so please check with Christine to confirm if this is acceptable for now until we do the refactor.  
image

**Review by RaminEsfandiari — CHANGES_REQUESTED** [review]

File: app/controllers/ReturnSentController.scala (hunk 2 above)
Comment (JakeReid2020) [reply app/controllers/ReturnSentController.scala:61]: Yeah these should be removed  later on in the refactor I think

File: test/controllers/ReturnSentControllerSpec.scala (hunk 4 above)
Comment (JakeReid2020) [reply test/controllers/ReturnSentControllerSpec.scala:564]: I'll ask  Christine but I think those are being moved in the refactor

**Review by RaminEsfandiari — APPROVED** [review]

**Pull request merged**